hie-plugin-api/Haskell/Ide/Engine/Context.hs

```diff
@@ -13,27 +15,86 @@ import Haskell.Ide.Engine.PluginUtils
 -- smarter code completion
 data Context = TypeContext
              | ValueContext
+             | ModuleContext String
+             | ImportContext String
+             | ImportListContext String
```

Comment:
What other information is relevant? List of imported symbols? 
Do we have to be careful to not clone the GHC/ghc-lib/haskell-src-exts API?

Reply:
Instead of trying to anticipate every possible use-case, I would leave it like that and required context information should be added as needed.